packages/application-components/src/components/custom-views/custom-view-loader/custom-view-loader.tsx

```diff
@@ -129,7 +150,14 @@ function CustomViewLoader(props: TCustomViewLoaderProps) {
   useEffect(() => {
     // Close the channel when the component unmounts
     const communicationChannel = iFrameCommunicationChannel.current;
+
+    // Listen for messages from the iFrame
+    iFrameCommunicationChannel.current!.port1.onmessage =
```

Comment:
We need to listen earlier. The Custom View will send the message in most cases before the iFrame is in ready state as the code already executed. If we do not move this earlier, then the host fails to react to the message.